### PR TITLE
Add unattended_url parameter

### PR DESCRIPTION
**Pull request by matonb**
Set `unattended_url` parameter in `foreman/settings.yaml`

Enables setting of unattended_url during install with foreman-installer

### CODE REVIEW

File: manifests/params.pp [hunk 1]
```diff
@@ -15,6 +15,7 @@
   $receive_facts  = true
   # should foreman manage host provisioning as well
   $unattended     = true
+  $unattended_url = "http://${lower_fqdn}:8000"
```
Comment (ekohl) [inline manifests/params.pp]: This is a non-default location I believe. When is it needed to override this and would it make sense to default to ```undef``` and only place it in the settings when the user overrides it?

Comment (matonb) [reply manifests/params.pp]: I can't check right now, but I'm pretty sure that's the current default location.

But sure, I was in two minds of setting it to a default location, or not at all...

Comment (matonb) [reply manifests/params.pp]: When would I use it?
Automated deployments of foreman puppet, my current need is to stand everything up from scratch with no user intervention

Comment (ekohl) [reply manifests/params.pp]: I don't recall having to ever override this and I did do bare metal deployments. Now that might just be my memory but I think by default the unattended url is based on the regular foreman url on port 80. A plain foreman doesn't even listen on port 8000 I think.

**Comment by matonb** [comment]
@ekohl Changed default to undef :)

**Comment by matonb** [comment]
@ekohl Want me to squash the commits?

**Comment by ekohl** [comment]
We can also do that on merge so you don't have to.

**Comment by matonb** [comment]
Cool

**Comment by mmoll** [comment]
a test would be nice.

**Comment by mmoll** [comment]
please rebase.

File: spec/classes/foreman_config_spec.rb [hunk 2]
```diff
@@ -161,6 +162,17 @@ class {'foreman':
...
 
+      describe 'with unattended_url' do
+        let :pre_condition do
+          "class {'foreman':
+            unattended_url => 'http://example.com',
+          }"
+        end
+
+        it { should contain_concat__fragment('foreman_settings+01-header.yaml').
+            with_content(/^:unattended_url:\s*http:\/\/example.com$/) }
```
Comment (ekohl) [inline spec/classes/foreman_config_spec.rb]: You can use ```with_content(%r{^:unattended_url:\s*http://example.com$})```

Comment (matonb) [reply spec/classes/foreman_config_spec.rb]: @ekohl 
  Thanks, I couldn't remember the regex pattern syntax and forgot to go back to it :)

**Comment by mmoll** [comment]
merged, thanks @matonb!

**Pull request merged**